core/src/main/java/hudson/model/UpdateCenter.java

```diff
@@ -1061,7 +1087,12 @@ public String getDisplayName() {
             return Messages.UpdateCenter_CoreUpdateMonitor_DisplayName();
         }
 
+        @Override
         public boolean isActivated() {
+            if (!Jenkins.get().getUpdateCenter().isSiteDataReady()) {
+                // Do not display monitor during this page load, but possibly later.
+                return false;
+            }
             Data data = getData();
```

Comment:
This is IMO not enough - it just masks the issue.  getData() will load from disk (uncached) - and this can be slow - and we should not do this to check.